sdk/src/main/java/com/mapbox/maps/MapProvider.kt

```diff
@@ -9,22 +9,20 @@ internal object MapProvider {
   fun getNativeMap(
     mapboxMapOptions: MapboxMapOptions,
     mapClient: MapClient,
-    mapObserver: MapObserver
   ): MapInterface = NativeMapImpl(
     Map(
       mapClient,
-      mapObserver,
       mapboxMapOptions.mapOptions,
       mapboxMapOptions.resourceOptions
     )
   )
 
   fun getMapboxMap(
     nativeMap: MapInterface,
-    mapObserver: NativeMapObserver,
+    nativeObserver: NativeObserver,
     pixelRatio: Float
   ) =
-    MapboxMap(nativeMap, mapObserver, pixelRatio)
+    MapboxMap(nativeMap, nativeObserver, pixelRatio)
```

Comment:
@pengdev could we remove `MapProvider` class at all? It's actually useless as it incapsulates nothing, just hides ctor invocations while each of that functions is used only once in project and once in test.

Reply:
I guess it's meant to help with testing, correct me if I'm wrong @tobrun